pkg/crypto/crypto_test.go

```diff
@@ -37,6 +37,75 @@ func TestGenerateSecp256k1Key(t *testing.T) {
 	}
 }
 
```

Comment:
becasue code seams to be repeating, this file would benefit with some helper function that would test EDG types and functions:

```
func testEdgFunc(t *testing.T, encodeFn, decodFn, generateFn) {
	t.Helper()
  
      // add testing code here
}

// then there could be test

func TestEncodeSecp256k1EDG(t *testing.T) {
	 t.Parallel()
	 
	edg := crypto.EDGSecp256_R1
	testEdgFunc(t, edg.Encode, edg.Decode, edg.Generate)
}

// repeat with other variations...
```